src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs

```diff
@@ -266,27 +266,51 @@ public static TextSpan MakeSpan(string text, int spanNum)
                 for (int i = 0; i < count; i++)
                 {
                     SourceText s = await proj.FindDocument(l[i]).GetTextAsync().ConfigureAwait(false);
-                    results.Add(s.ToString().Replace("\r\n", "\n", StringComparison.Ordinal));
+                    results.Add(Replace(s.ToString(), "\r\n", "\n"));
                 }
             }
             else
             {
                 for (int i = 0; i < count; i++)
                 {
                     SourceText s = await proj.FindDocument($"src-{i}.cs").GetTextAsync().ConfigureAwait(false);
-                    results.Add(s.ToString().Replace("\r\n", "\n", StringComparison.Ordinal));
+                    results.Add(Replace(s.ToString(), "\r\n", "\n"));
                 }
             }
 
             if (extraFile != null)
             {
                 SourceText s = await proj.FindDocument(extraFile).GetTextAsync().ConfigureAwait(false);
-                results.Add(s.ToString().Replace("\r\n", "\n", StringComparison.Ordinal));
+                results.Add(Replace(s.ToString(), "\r\n", "\n"));
             }
 
             return results;
         }
 
+        public static bool CompareLines(string[] expectedLines, SourceText sourceText, out string message)
```

Comment:
💡 `AssertEx.EqualOrDiff` is pretty magical if you can bring it over from either dotnet/roslyn or dotnet/roslyn-sdk.